rust/main/agents/relayer/src/prover.rs

```diff
@@ -11,7 +11,7 @@ use tracing::{error, instrument};
 
 /// A depth-32 sparse Merkle tree capable of producing proofs for arbitrary
 /// elements.
-#[derive(Debug)]
```

Comment:
I wonder if we can actually clone the prover. Did we use a single instance beforehand?